src/community/modularity.c

```diff
@@ -216,6 +216,46 @@ int igraph_modularity(const igraph_t *graph,
     return IGRAPH_SUCCESS;
 }
 
+static int igraph_i_modularity_matrix_get_adjacency(
+           const igraph_t *graph, igraph_matrix_t *res,
+           const igraph_vector_t *weights, igraph_bool_t directed) {
+    /* Specifically used to handle weights and/or ignore direction */
+    igraph_eit_t edgeit;
+    long int no_of_nodes = igraph_vcount(graph);
+    igraph_integer_t from, to;
+
+    IGRAPH_CHECK(igraph_matrix_resize(res, no_of_nodes, no_of_nodes));
+    igraph_matrix_null(res);
+    IGRAPH_CHECK(igraph_eit_create(graph, igraph_ess_all(0), &edgeit));
+    IGRAPH_FINALLY(igraph_eit_destroy, &edgeit);
+
+    if (weights) {
+        while (!IGRAPH_EIT_END(edgeit)) {
+            igraph_integer_t edge = IGRAPH_EIT_GET(edgeit);
+            igraph_edge(graph, edge, &from, &to);
+            MATRIX(*res, from, to) += VECTOR(*weights)[edge];
+            if (!directed) {
+                MATRIX(*res, to, from) += VECTOR(*weights)[edge];
+            }
+        IGRAPH_EIT_NEXT(edgeit);
```

Comment:
Indentation is inconsistent here.

With edge/vertex iterators, one can also use `for`:

```
for (; !IGRAPH_EITEND(edgeit); IGRAPH_EIT_NEXT(edgeit))
```

This is a stylistic matter and it is up to you. I mentioned it in case you used this indentation on purpose, to highlight the last line. In that case, `for` would be better.